### PR TITLE
[ACR] Add Retention Policy commands

### PR DESCRIPTION
**Pull request by ankurkhemani**
Add commands for `az acr config retention`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by ankurkhemani** [comment]
@yugangw-msft Can you please add it to [Sprint 73](https://github.com/Azure/azure-cli/milestone/72)?
@djyou @nathana1  Can you please help review?

File: src/azure-cli/azure/cli/command_modules/acr/policy.py [hunk 1]
```diff
@@ -43,3 +43,45 @@ def acr_config_content_trust_update(cmd,
...
+                                client,
+                                registry_name,
+                                status=None,
+                                days=None,
+                                resource_group_name=None):
+    registry, resource_group_name = validate_premium_registry(
+        cmd, registry_name, resource_group_name, POLICIES_NOT_SUPPORTED)
+
+    Policy = cmd.get_models('Policy')
+    RetentionPolicy = cmd.get_models('RetentionPolicy')
```
[the remark this answers is not in the document]
Comment (ankurkhemani) [reply src/azure-cli/azure/cli/command_modules/acr/policy.py]: Done.

File: src/azure-cli/azure/cli/command_modules/acr/policy.py [hunk 2]
```diff
@@ -43,3 +43,45 @@ def acr_config_content_trust_update(cmd,
...
+
+    policies = registry.policies
+    if policies is None:
+        policies = Policy()
+
+    policies.retention_policy = policies.retention_policy if policies.retention_policy else RetentionPolicy()
+
+    if status is not None:
```
[the remark this answers is not in the document]
Comment (ankurkhemani) [reply src/azure-cli/azure/cli/command_modules/acr/policy.py]: Changed.

File: src/azure-cli/azure/cli/command_modules/acr/_params.py [hunk 3]
```diff
@@ -76,9 +76,12 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('repository', help='The repository name for a manifest-only copy of images. Multiple copies supported by passing --repository multiple times.', action='append')
         c.argument('force', help='Overwrite the existing tag of the image to be imported.', action='store_true')
 
-    with self.argument_context('acr config content-trust') as c:
+    with self.argument_context('acr config') as c:
         c.argument('status', help="Indicates whether content-trust is enabled or disabled.", arg_type=get_enum_type(PolicyStatus))
 
+    with self.argument_context('acr config retention') as c:
+        c.argument('days', help='The number of days to retain an untagged manifest after which it gets purged. Defaults to: 7')
```
[the remark this answers is not in the document]
Comment (ankurkhemani) [reply src/azure-cli/azure/cli/command_modules/acr/_params.py]: Done.

**Comment by ankurkhemani** [comment]
@yugangw-msft There are some failing sql tests? Can you please check?

**Comment by yugangw-msft** [comment]
> @yugangw-msft There are some failing sql tests? Can you please check?

Sure, we are looking for a fix right now. azure-mgmt-sql had a new release yesterday, which exposes a dependency issue at CLI end.

**Comment by ankurkhemani** [comment]
@yugangw-msft Thanks! This PR is ready to merge once the CI tests are fixed.

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py [hunk 4]
```diff
@@ -78,3 +79,10 @@ def validate_task_argument(string, is_secret):
             return {'type': 'Argument', 'name': comps[0], 'value': local_value, 'isSecret': is_secret}
         return {'type': 'Argument', 'name': comps[0], 'value': '', 'isSecret': is_secret}
     return None
+
+
+def validate_retention_days(namespace):
+    days = namespace.days
+    if isinstance(days, int):
```
Comment (djyou) [inline src/azure-cli/azure/cli/command_modules/acr/_validators.py]: Is this check needed?

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py [hunk 5]
```diff
@@ -78,3 +79,10 @@ def validate_task_argument(string, is_secret):
...
+        if days and (days < 0 or days > 365):
+            raise CLIError("invalid value for days: should be from 0 to 365")
```
Comment (djyou) [inline src/azure-cli/azure/cli/command_modules/acr/_validators.py]: Invalid

Comment (djyou) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: nit: no strong opinion here but prefer leaving the validation to the server side if possible.

Comment (ankurkhemani) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: RP will add the validation in parallel. Don't want to block the release for that or not have the validation.

File: src/azure-cli/azure/cli/command_modules/acr/_validators.py (hunk 4 above)
Comment (yugangw-msft) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: this is probably needed, or more particularly would be "days is not None"

Comment (ankurkhemani) [reply src/azure-cli/azure/cli/command_modules/acr/_validators.py]: removed.

**Pull request merged**